### PR TITLE
Add testcase index to CWS/AWS submission details

### PR DESCRIPTION
**Pull request by wil93**
CWS before:

CWS before

CWS after:

CWS after

AWS before:

AWS before

AWS after:

AWS after

[Review on Reviewable](https://reviewable.io/reviews/cms-dev/cms/541)

### CODE REVIEW

**Comment by stefano-maggiolo** [comment]
Ok, but you don't want to disclose the testcase codename. Please use an actual index (I would vote for 1-based, honestly).

---

Review status: 0 of 5 files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cms-dev/cms/541#-:-KAag71K6z3oHx4lR47b:-2135081222)*

File: cms/server/admin/templates/fragments/submission_row.html [hunk 1]
```diff
@@ -46,6 +46,7 @@
         <table class="nested bordered">
           <thead>
             <tr>
+              <th>#</th>
```
Comment (wil93) [inline cms/server/admin/templates/fragments/submission_row.html]: For some reason this file (`cms/server/admin/templates/fragments/submission_row.html`) doesn't need to be patched. I think that the table is taken from the ScoreType HTML template. I am not sure if this file is even used anymore.

May we just remove this file?

/cc @stefano-maggiolo

**Comment by stefano-maggiolo** [comment]
Review status: 0 of 5 files reviewed at latest revision, 1 unresolved discussion.

---

*[cms/server/admin/templates/fragments/submission_row.html, line 49 \[r1\]](https://reviewable.io:443/reviews/cms-dev/cms/541#-KAafgN44BRmoRbSKY4V-r1-49:-KAaiWzTdavj9fY0MAye:-2068322963) ([raw file](https://github.com/cms-dev/cms/blob/c359cb80b3cec2e6cfa50cbb1ee95e45e3bbe7f3/cms/server/admin/templates/fragments/submission_row.html#L49)):*
It is used by the submission_rows fragment, which in turn is used by everything that displays submission tables in AWS. Or am I missing something?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cms-dev/cms/541)*

**Comment by wil93** [comment]
Review status: 0 of 5 files reviewed at latest revision, 1 unresolved discussion.

---

*[cms/server/admin/templates/fragments/submission_row.html, line 49 \[r1\]](https://reviewable.io:443/reviews/cms-dev/cms/541#-KAafgN44BRmoRbSKY4V-r1-49:-KAay_AyjNYEVy3uSC5a:2097488442) ([raw file](https://github.com/cms-dev/cms/blob/c359cb80b3cec2e6cfa50cbb1ee95e45e3bbe7f3/cms/server/admin/templates/fragments/submission_row.html#L49)):*
I'm not sure, but I think the submission table is generated by the scoretype (in fact it has to be, since this file does not support subtasks).

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cms-dev/cms/541)*

**Comment by wil93** [comment]
Review status: 0 of 5 files reviewed at latest revision, 1 unresolved discussion.

---

*[cms/server/admin/templates/fragments/submission_row.html, line 49 \[r1\]](https://reviewable.io:443/reviews/cms-dev/cms/541#-KAafgN44BRmoRbSKY4V-r1-49:-KAb3BCcY1TNvfDZDVPx:-166793761) ([raw file](https://github.com/cms-dev/cms/blob/c359cb80b3cec2e6cfa50cbb1ee95e45e3bbe7f3/cms/server/admin/templates/fragments/submission_row.html#L49)):*
Actually, this file is used. It's just this particular section that it's only used when the submission is in "Scoring..." status. Why do we want to show the outcomes before the scoring has finished?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cms-dev/cms/541)*

**Comment by wil93** [comment]
Done. Also changed it in another part.

Before: 


After: 


---

Review status: 0 of 6 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cms-dev/cms/541#-:-KAb47OQaDAXE2gB2PN_:244153581)*

**Comment by stefano-maggiolo** [comment]
Reviewed 3 of 3 files at r2.
Review status: 3 of 6 files reviewed at latest revision, 3 unresolved discussions.

---

*[cms/grading/ScoreType.py, line 206 \[r2\]](https://reviewable.io:443/reviews/cms-dev/cms/541#-KBA9vZKUK7byynmR6Iz:-KBA9vZKUK7byynmR6J-:-1538408596) ([raw file](https://github.com/cms-dev/cms/blob/6abafcd9b6eee1c2f09a197ee90aac4b065021be/cms/grading/ScoreType.py#L206)):*
Can you do `for idx, tc in enumerate(st["testcases"])` (consequently, idx + 1 later) and save the set which is hideous?

---

*[cms/grading/scoretypes/Sum.py, line 63 \[r2\]](https://reviewable.io:443/reviews/cms-dev/cms/541#-KBAA7A_5U6dtG4eNyN_:-KBAA7A_5U6dtG4eNyNa:-448103517) ([raw file](https://github.com/cms-dev/cms/blob/6abafcd9b6eee1c2f09a197ee90aac4b065021be/cms/grading/scoretypes/Sum.py#L63)):*
Ditto here.

---

*[cms/server/admin/templates/fragments/submission_row.html, line 49 \[r1\]](https://reviewable.io:443/reviews/cms-dev/cms/541#-KAafgN44BRmoRbSKY4V-r1-49:-KBAAAGe5lAQa0msZ2pC:-1455979682) ([raw file](https://github.com/cms-dev/cms/blob/c359cb80b3cec2e6cfa50cbb1ee95e45e3bbe7f3/cms/server/admin/templates/fragments/submission_row.html#L49)):*
Why not? If SS is not running, we'd like to show admins what we got.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cms-dev/cms/541)*

**Comment by wil93** [comment]
Review status: 3 of 6 files reviewed at latest revision, 2 unresolved discussions.

---

*[cms/grading/ScoreType.py, line 206 \[r2\]](https://reviewable.io:443/reviews/cms-dev/cms/541#-KBA9vZKUK7byynmR6Iz:-KBAGRWCZ0XNJNyhxv4l:-609051917) ([raw file](https://github.com/cms-dev/cms/blob/6abafcd9b6eee1c2f09a197ee90aac4b065021be/cms/grading/ScoreType.py#L206)):*
I think we can't do it like that here, because `idx` would always start from 1 inside subtasks.

---

*[cms/grading/scoretypes/Sum.py, line 63 \[r2\]](https://reviewable.io:443/reviews/cms-dev/cms/541#-KBAA7A_5U6dtG4eNyN_:-KBAGQpcJG1eHZJ7NnMk:-499146009) ([raw file](https://github.com/cms-dev/cms/blob/6abafcd9b6eee1c2f09a197ee90aac4b065021be/cms/grading/scoretypes/Sum.py#L63)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cms-dev/cms/541)*

**Comment by stefano-maggiolo** [comment]
Reviewed 3 of 5 files at r1, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[cms/grading/scoretypes/Sum.py, line 70 \[r3\]](https://reviewable.io:443/reviews/cms-dev/cms/541#-KBCo1Xh_lfmK1OhSl6l:-KBCo1Xh_lfmK1OhSl6m:-472822540) ([raw file](https://github.com/cms-dev/cms/blob/895e184c80de1370076e1dc9df81a4fb9e93c3b6/cms/grading/scoretypes/Sum.py#L70)):*
idx + 1

---

*[cms/server/admin/templates/fragments/submission_row.html, line 58 \[r3\]](https://reviewable.io:443/reviews/cms-dev/cms/541#-KBCo3hI23bMrRyMLzxy:-KBCo3hI23bMrRyMLzxz:-522726196) ([raw file](https://github.com/cms-dev/cms/blob/895e184c80de1370076e1dc9df81a4fb9e93c3b6/cms/server/admin/templates/fragments/submission_row.html#L58)):*
Confusion: this idx is really the codename, can you change the name of this variable, and add the "real" idx as in all the other parts?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cms-dev/cms/541)*

**Comment by wil93** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[cms/grading/scoretypes/Sum.py, line 70 \[r3\]](https://reviewable.io:443/reviews/cms-dev/cms/541#-KBCo1Xh_lfmK1OhSl6l:-KBDcrzIBiwRf8y-KSbI:-1211756979) ([raw file](https://github.com/cms-dev/cms/blob/895e184c80de1370076e1dc9df81a4fb9e93c3b6/cms/grading/scoretypes/Sum.py#L70)):*
It should already be 1-based (I did `enumerate(details, start=1)`)

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cms-dev/cms/541)*

**Comment by wil93** [comment]
Review status: 5 of 7 files reviewed at latest revision, 2 unresolved discussions.

---

*[cms/server/admin/templates/fragments/submission_row.html, line 58 \[r3\]](https://reviewable.io:443/reviews/cms-dev/cms/541#-KBCo3hI23bMrRyMLzxy:-KBDm8oP-LKKLyWN4-n2:430330326) ([raw file](https://github.com/cms-dev/cms/blob/895e184c80de1370076e1dc9df81a4fb9e93c3b6/cms/server/admin/templates/fragments/submission_row.html#L58)):*
I just changed it here (and in a couple other places) by using the [inverse `zip` function](http://stackoverflow.com/questions/19339/a-transpose-unzip-function-in-python-inverse-of-zip) to obtain just the `ev`s, and wrapping them with an `enumerate(..., start=1)` call

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cms-dev/cms/541)*

**Comment by stefano-maggiolo** [comment]
Sorry for the many back and forth - all right now apart from the "visible to contestant" cell you removed and that other nit.

---

Reviewed 1 of 2 files at r4, 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[cms/grading/scoretypes/Sum.py, line 70 \[r3\]](https://reviewable.io:443/reviews/cms-dev/cms/541#-KBCo1Xh_lfmK1OhSl6l:-KBH2WNgEArjfGR0T73g:-1828983883) ([raw file](https://github.com/cms-dev/cms/blob/895e184c80de1370076e1dc9df81a4fb9e93c3b6/cms/grading/scoretypes/Sum.py#L70)):*
Sorry, did not see.

---

*[cms/server/admin/templates/fragments/submission_row.html, line 62 \[r5\]](https://reviewable.io:443/reviews/cms-dev/cms/541#-KBH2nPuhQY3hhAqgUo1:-KBH2nPvMhAXb4LmK9Ox:-1816833280) ([raw file](https://github.com/cms-dev/cms/blob/64602bd8d2002a5f712881cf139d27d72ebcfe54/cms/server/admin/templates/fragments/submission_row.html#L62)):*
Why did this disappear?

---

*[cms/server/admin/templates/task.html, line 446 \[r5\]](https://reviewable.io:443/reviews/cms-dev/cms/541#-KBH3FUV1rBaO54qf4GD:-KBH3FUV1rBaO54qf4GE:-1929634333) ([raw file](https://github.com/cms-dev/cms/blob/88c34e15be309ad078dbbb3f5ac62fdd85b99f6f/cms/server/admin/templates/task.html#L446)):*
Indent 2

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cms-dev/cms/541#-:-KBH3HvNDfr4uIF5OCna:-1289957053)*

**Comment by wil93** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[cms/server/admin/templates/fragments/submission_row.html, line 62 \[r5\]](https://reviewable.io:443/reviews/cms-dev/cms/541#-KBH2nPuhQY3hhAqgUo1:-KBIGm6jzemDVwtzM45g:-324956263) ([raw file](https://github.com/cms-dev/cms/blob/64602bd8d2002a5f712881cf139d27d72ebcfe54/cms/server/admin/templates/fragments/submission_row.html#L62)):*
I thought it wasn't needed, but now I understand that the "bullet" was basically a true/false indicator... I'll add it back (I'll try to make it more clear)

---

*[cms/server/admin/templates/task.html, line 446 \[r5\]](https://reviewable.io:443/reviews/cms-dev/cms/541#-KBH3FUV1rBaO54qf4GD:-KBIGYSX2Fl4Ey_OwcKu:541040248) ([raw file](https://github.com/cms-dev/cms/blob/88c34e15be309ad078dbbb3f5ac62fdd85b99f6f/cms/server/admin/templates/task.html#L446)):*
Do you mean just `set` or everything inside `for`-`end`?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cms-dev/cms/541)*

**Comment by stefano-maggiolo** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[cms/server/admin/templates/task.html, line 446 \[r5\]](https://reviewable.io:443/reviews/cms-dev/cms/541#-KBH3FUV1rBaO54qf4GD:-KBILT1qx2q0b9dYsNhV:-1029523609) ([raw file](https://github.com/cms-dev/cms/blob/88c34e15be309ad078dbbb3f5ac62fdd85b99f6f/cms/server/admin/templates/task.html#L446)):*
Just set... We don't have precise rules on indenting tornado templates, my hunch is to try to be locally consistent for each language (python/html)

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cms-dev/cms/541)*

**Comment by wil93** [comment]
Review status: 4 of 7 files reviewed at latest revision, 2 unresolved discussions.

---

*[cms/server/admin/templates/task.html, line 446 \[r5\]](https://reviewable.io:443/reviews/cms-dev/cms/541#-KBH3FUV1rBaO54qf4GD:-KBI_gNQrvDcvM0UdaxN:-499146009) ([raw file](https://github.com/cms-dev/cms/blob/88c34e15be309ad078dbbb3f5ac62fdd85b99f6f/cms/server/admin/templates/task.html#L446)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cms-dev/cms/541)*

**Comment by stefano-maggiolo** [comment]
Reviewed 3 of 3 files at r6.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cms-dev/cms/541)*

**Comment by stefano-maggiolo** [comment]
Merged, with a small change for the tests to pass (the id="eval_x_codename" were not useless).

**Pull request closed without merging**